vault/resource_pki_secret_backend_crl_config.go

```diff
@@ -0,0 +1,118 @@
+package vault
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/vault/api"
+)
+
+func pkiSecretBackendCrlConfigResource() *schema.Resource {
+	return &schema.Resource{
+		Create: pkiSecretBackendCrlConfigCreate,
+		Read:   pkiSecretBackendCrlConfigRead,
+		Update: pkiSecretBackendCrlConfigUpdate,
+		Delete: pkiSecretBackendCrlConfigDelete,
+
+		Schema: map[string]*schema.Schema{
+			"backend": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The path of the PKI secret backend the resource belongs to.",
+			},
```

Comment:
A lot of endpoints leave this as optional and default it to the engine name. If we went that way, this is probably what it'd look like:
```
"backend": {
	Type:        schema.TypeString,
	Optional:    true,
	Description: "The path of the PKI secret backend the resource belongs to.",
	ForceNew:    true,
	Default:     "pki",
	// standardise on no beginning or trailing slashes
	StateFunc: func(v interface{}) string {
		return strings.Trim(v.(string), "/")
	},
},
```
Do you think we should make it more like that, or are there reasons it's better as-is?

Reply:
I have added half of your suggestion here, but "Optional" wouldn't work in conjunction with "Default"